src/lazy/encoder/binary/v1_1/container_writers.rs

```diff
@@ -110,20 +110,76 @@ impl<'value, 'top> SequenceWriter for BinarySExpWriter_1_1<'value, 'top> {
 }
 
 pub struct BinaryStructWriter_1_1<'value, 'top> {
+    // When true, struct field names are encoded as `FlexUInt`s.
+    // When false, struct field names are encoded as `FlexSym`s.
+    //
+    // Always starts as `true`; remains `true` as long as field names being written
+    // are symbol IDs. Once a field name with inline text needs to be encoded, switches to `false`.
+    flex_uint_encoding: bool,
     container_writer: BinaryContainerWriter_1_1<'value, 'top>,
 }
 
 impl<'value, 'top> BinaryStructWriter_1_1<'value, 'top> {
     pub(crate) fn new(container_writer: BinaryContainerWriter_1_1<'value, 'top>) -> Self {
-        Self { container_writer }
+        Self {
+            flex_uint_encoding: true,
+            container_writer,
+        }
+    }
+
+    fn enable_flex_sym_encoding(&mut self) {
+        if self.flex_uint_encoding {
+            // Write a zero byte out to signal to future readers that we are switching from
+            // FlexUInt to FlexSym.
+            self.container_writer.buffer().push(0x00);
```

Comment:
In Java we ended up using FlexInt zero (0x01) instead of 0x00. See https://github.com/amazon-ion/ion-java/pull/673/files

Reply:
In Java, if nothing had been written yet we swapped out the opcode (to 0xFD: variable length, FlexSym fields) instead of writing the switch byte. I think the idea is that if you're using FlexSyms you're more likely to exceed the lower nibble anyway, so you can use the byte for the length instead and potentially save a byte. We can discuss whether this is worthwhile though. If it's not, then maybe 0xFD can be reclaimed. See https://github.com/amazon-ion/ion-java/pull/665/files#diff-1903a8a68f333503cde63de27c671d623ff68da4accf35aa1db835d2c3e6f783R339